images/win/scripts/Installers/Install-Msys2.ps1

```diff
@@ -11,21 +11,25 @@ $origPath = $env:PATH
 $gitPath  = "$env:ProgramFiles\Git"
 
 # get info from https://sourceforge.net/projects/msys2/files/Base/x86_64/
-$msys2Uri  = "http://repo.msys2.org/distrib/x86_64/msys2-base-x86_64-20190524.tar.xz"
+$msys2Uri  = "https://github.com/msys2/msys2-installer/releases/download/2020-05-17/msys2-base-x86_64-20200517.tar.xz"
```

Comment:
@MSP-Greg , what do you think about reworking it to consume the latest release always?
```
$latestRelease = Invoke-RestMethod "https://api.github.com/repos/msys2/msys2-installer/releases/latest"
$downloadAsset = $latestRelease.assets | Where-Object { $_.name -match "x86_64" -and $_.name -match "tar.xz" }                         
$msys2Uri = $downloadAsset.browser_download_url
```

Reply:
Good idea.  Used REST from a lot of languages, but never PS...  Collapsed a bit, hate needing to horizontally scroll code windows...

Reply:
An alternative is using the one explicitly marked as the latest, these should be the same 

https://github.com/msys2/msys2-installer/releases/download/nightly-x86_64/msys2-base-x86_64-latest.tar.xz

Reply:
I don't think that we should install nightly build to image.

Reply:
The two tarballs are the same exact files.

```bash
sha256sum msys2-base-x86_64-20200517.tar.xz msys2-base-x86_64-latest.tar.xz
c4443113497acb2d2e285d40b929fc55f33f8f669902595ecdf66a655b63dc60  msys2-base-x86_64-20200517.tar.xz
c4443113497acb2d2e285d40b929fc55f33f8f669902595ecdf66a655b63dc60  msys2-base-x86_64-latest.tar.xz
```

Reply:
![image](https://user-images.githubusercontent.com/16715858/82465985-87c20e80-9ac8-11ea-97d4-409fa1088838.png)

Reply:
> The two tarballs are the same exact files.

In a month that may not be the case.  Although not common, there have been reversions in some packages.  Whether that has happened with any 'base' packages, I don't recall.  With the update code, both will yield the same set of packages?

@lazka sorry for the ping.  For the starting msys2 package for Actions Windows images, would you suggest using

msys2-base-x86_64-20200517.tar.xz

or

msys2-base-x86_64-latest.tar.xz?

Reply:
https://gitter.im/msys2/msys2?at=5ec5526e9832dd6f0464673c
![image](https://user-images.githubusercontent.com/8345542/82468846-e6679800-9a88-11ea-9462-2791d37a1063.png)

https://gitter.im/msys2/msys2?at=5ec5539bcf55346b5fbca538
![image](https://user-images.githubusercontent.com/8345542/82469058-23338f00-9a89-11ea-8ddf-795fa5428286.png)

From lazka

Reply:
@1480c1,
I don't see `msys2-base-x86_64-latest.tar.xz` file in `latest` release: https://api.github.com/repos/msys2/msys2-installer/releases/latest
Is there any reason to use `nightly-x86_64` tag instead of `latest`?

Reply:
`msys2-base-x86_64-latest.tar.xz` is in the [`nightly-x86_64`](https://github.com/msys2/msys2-installer/releases/tag/nightly-x86_64) release

One reason to use `nightly-x86_64` is to reduce the number of packages that would need to be updated, depending on how behind the `latest` release is

Reply:
I am not familiar with MSYS2 release scheme but usually stable releases of tool are distributed as a separate GitHub releases, like this one: https://github.com/msys2/msys2-installer/releases/tag/2020-05-17
As I can see, Nightly builds are built based on master branch and just update the single release: https://github.com/msys2/msys2-installer/releases/tag/nightly-x86_64 (it means no changelog, no updates history)

And as it is mentioned in repository, those builds are not tested.
As [author mentioned](https://gitter.im/msys2/msys2?at=5ec5539bcf55346b5fbca538), they publish stable releases as separate tags after "manual testing".

It means that any bug introduced by new changes, will be picked up by nightly build and then will be consumed during image generation.